src/cargo/sources/registry/mod.rs

```diff
@@ -548,12 +550,17 @@ impl<'cfg> RegistrySource<'cfg> {
         // After we've loaded the package configure its summary's `checksum`
         // field with the checksum we know for this `PackageId`.
         let req = VersionReq::exact(package.version());
-        let summary_with_cksum = self
-            .index
-            .summaries(package.name(), &req, &mut *self.ops)?
-            .map(|s| s.summary.clone())
-            .next()
-            .expect("summary not found");
+        let summary_with_cksum = loop {
+            match self.index.summaries(package.name(), &req, &mut *self.ops)? {
+                Poll::Ready(summaries) => {
+                    break summaries;
+                }
+                Poll::Pending => (), // TODO: dont hot loop for it to be Ready
```

Comment:
Should we just return `Pending` here too?

Reply:
Either that or I think it should be safe to assert at this point since we should be done downloading.

Reply:
Both parents already have loops, so returning `Pool` was tempting. But both do things that did not look like they wanted to be called in a loop (`download` and `finish_download`) so used an assert.